### PR TITLE
Lightbox 2.0: add transition for caption

### PR DESCRIPTION
**Pull request by muxin**
Partial for https://github.com/ampproject/amphtml/issues/9970

### CODE REVIEW

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 1]
```diff
@@ -239,29 +242,69 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
-          if (state.descTextAreaHeight < state.descBoxHeight) {
-            const diffHeight = state.descBoxHeight - state.descTextAreaHeight;
-            setStyle(this.descriptionTextArea_, 'top', `${diffHeight}px`);
-          }
+        mutate: state1 => {
+          this.descriptionTextArea_.classList.remove('non-expanded');
+          const tempDiffHeight =
+              state1.descBoxHeight - state1.prevDescTextAreaHeight;
+          setStyle(this.descriptionTextArea_, 'top', `${tempDiffHeight}px`);
+          this.vsync_.run({
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: let's move the code inside this `run()` to a constant outside for better readability i.e. something like `this.vsync_.run(animate)`

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 2]
```diff
@@ -239,29 +242,69 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
+            measure: state => {
+              state.descTextAreaHeight =
+                  this.descriptionTextArea_./*OK*/scrollHeight;
+              state.descBoxHeight = this.descriptionBox_./*OK*/clientHeight;
+            },
+            mutate: state => {
+              const finalDiffHeight =
+                  state.descBoxHeight - state.descTextAreaHeight;
+              if (finalDiffHeight > 0) {
+                this.animateDescOverflow_(
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: move params to variables so we can do a single `this.animateDescOverflow_` call.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js (hunk 1 above)
Comment (muxin) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: done

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js (hunk 2 above)
Comment (muxin) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: done

**Pull request merged**